### ISSUE TITLE
Feature Request: Allow non-integer values for scale.

### ISSUE THREAD

**Issue by SeanHRN** [GDQuest/krita-batch-exporter#61]
**I'm submitting a...**

- [ ] Bug report.
- [x ] Feature request.

## Feature request ##

**Describe the problem you're trying to solve.**
I would like for non-integer values to be accepted as export percentages.

I am a motion comic developer who works between Krita and Ren'Py. This batch exporter has been an excellent addition to my workflow. I wrote a plugin that generates Ren'Py scripting based on how the images appear on the canvas and how the layers are named with your plugin's syntax:
[https://github.com/SeanHRN/generate-renpy-scripting](url)

My target resolution is 1920x1080, because that's the Ren'Py standard. The one problem is that this batch exporter doesn't accept non-integer values for the export's scale percentage. This means that in order to ensure that I could get my image to reach a height of exactly 1080, my document's height must be (1080 * 2 * N)px. If I want to make the canvas a reasonable 5760x3240px, I would have to be willing to re-import the exported images into Krita and then scale them down manually just because the required scale would be 33.33%.

While my plugin itself allows non-integer scales from the layer names, its current design discourages that because it has to follow the standards of your batch exporter; there's a scale percentage calculator to preview the export size at different percentages, and it's currently integer-only.

**Comment by NathanLovato** [comment]
The addon should already support that, like -s 0.3333? Or doesn't it? @razcore-rad do you remember if the addon only supports whole numbers for scale?

**Comment by SeanHRN** [comment]
-s 0.33 just exports a copy with no scaling.
s=0.33 gave me this:
ERROR: invalid literal for int() with base 10: '0.33'
I'll take a look at the code.

**Comment by SeanHRN** [comment]
Looks like this works in def meta() in the WNode class:
```
meta.update({k: list(map(float, v)) for k, v in meta.items() if k in "s"})
meta.update({k: list(map(int, v)) for k, v in meta.items() if k in "m"})
```

**Comment by NathanLovato** [comment]
Ah, that's because the scale works in base 100, so 50 is 50% and so on. I remember now. Thanks for the pr, well check it later

**Comment by razcore-rad** [comment]
Closed via [6918f55](https://github.com/GDQuest/krita-batch-exporter/commit/6918f55290ae46d098e6293d874c5f9ab5f1b7ae)
Thanks again.
